src/unstract/sdk/exceptions.py

```diff
@@ -12,5 +12,6 @@ def __str__(self) -> str:
 
 class IndexingError(SdkError):
     def __init__(self, message: str = ""):
-        prefix = "Error with indexing. "
-        super().__init__(prefix + message)
+        if "404" in message:
```

Comment:
Unsure if this is the best way to check - as the presence of a string `404` anywhere in the message will cause it to be overridden.

Since this is a 404 from some internal library, we could ideally have a separate `IndexNotFoundError` which can be caught by the `backend` and eventually return 404 error code itself

Reply:
@chandrasekharan-zipstack Most importantly we do not need a "Error with indexing" prefix when the type of exception itself is `IndexingError`.

Reply:
@hari-kuriakose I'm okay with this assumption until we hit a blocker.
One benefit that we might miss out on by not explicitly differentiating `IndexingError` and `IndexNotFoundError` is that the BE can be clever enough to return `404` error code in case of the latter.
Either way, I think in the existing system we return 400 / 500 which needs to be looked into. Since it pertains to the `unstract` repo we can address it separately